Source/WebKit/WebProcess/GPU/media/SourceBufferPrivateRemote.cpp

```diff
@@ -81,17 +81,26 @@ SourceBufferPrivateRemote::~SourceBufferPrivateRemote()
     gpuProcessConnection->messageReceiverMap().removeMessageReceiver(Messages::SourceBufferPrivateRemote::messageReceiverName(), m_remoteSourceBufferIdentifier.toUInt64());
 }
 
-void SourceBufferPrivateRemote::append(Ref<SharedBuffer>&& data)
+Ref<GenericPromise> SourceBufferPrivateRemote::append(Ref<SharedBuffer>&& data)
 {
     auto gpuProcessConnection = m_gpuProcessConnection.get();
     if (!isGPURunning())
-        return;
+        return GenericPromise::createAndReject(-1);
+
+    return gpuProcessConnection->connection().sendWithPromisedReply(Messages::RemoteSourceBufferProxy::Append(IPC::SharedBufferReference { WTFMove(data) }), m_remoteSourceBufferIdentifier)->whenSettled(RunLoop::current(), [weakThis = WeakPtr { *static_cast<SourceBufferPrivate*>(this) }, this](auto&& result) {
```

Comment:
Why is `* static_cast<SourceBufferPrivate*>(this)` necessary?

Reply:
Because SourceBufferPrivateRemote inherit from both `SourceBufferPrivate` and `IPC::MessageReceiver` and both inherit from CanMakeWeakPtr ; as such, when building the WeakPtr, the compiler can't decide if it will use SourceBufferPrivate::weakPtrFactory() or MessageReceiver::weakPtrFactory() and it needs to be disambiguated.